numba/numpy_support.py

```diff
@@ -87,20 +87,28 @@ def from_dtype(dtype):
     Return a Numba Type instance corresponding to the given Numpy *dtype*.
     NotImplementedError is raised on unsupported Numpy dtypes.
     """
-    if dtype.fields is None:
+    if type(dtype) == type and issubclass(dtype, np.generic):
+        dtype = np.dtype(dtype)
+    elif getattr(dtype, "fields", None) is not None:
+        return from_struct_dtype(dtype)
+
+    try:
+        return FROM_DTYPE[dtype]
+    except KeyError:
         try:
-            return FROM_DTYPE[dtype]
-        except KeyError:
-            if dtype.char in 'SU':
+            char = getattr(dtype, 'char')
+        except AttributeError:
```

Comment:
i'm curious.  what dtype object is missing the `char` attribute?

Reply:
Changed to just reference the dtype.char attribute.